src/server/page_server.cpp

```diff
@@ -204,18 +204,80 @@ page_server::connection_handler::receive_start_catch_up (tran_server_conn_t::seq
   unpacker.unpack_int (port);
   cublog::lsa_utils::unpack (unpacker, catchup_lsa);
 
-  er_log_debug (ARG_FILE_LINE, "receive_start_catch_up: hostname = %s, port = %d, LSA = (%lld|%d)\n", host.c_str (), port,
-		LSA_AS_ARGS (&catchup_lsa));
+  _er_log_debug (ARG_FILE_LINE, "[CATCH-UP] Start catching up with the follower (%s:%d), until LSA = (%lld|%d)\n",
+		 host.c_str (), port,LSA_AS_ARGS (&catchup_lsa));
   if (port == -1)
     {
       return; // TODO: It means that the ATS is booting up, it will be set properly after ATS recovery is implemented.
     }
 
   // TODO: A thread will take the catch-up including establishing connection to avoid blocking ATS->PS reqeusts.
-  m_ps.connect_to_followee_page_server (std::move (host), port);
+  // All the below here will be taken up.
+  {
+    m_ps.connect_to_followee_page_server (std::move (host), port);
+
+    assert (!log_Gl.hdr.append_lsa.is_null ());
+    assert (!catchup_lsa.is_null ());
+    assert (catchup_lsa >= log_Gl.hdr.append_lsa);
+
+    if (log_Gl.hdr.append_lsa == catchup_lsa)
```

Comment:
@wrlawodms (post-merge comment)
Actually this decision can be done before connecting to the followee page server.

From an ATS's perspective there are 3 functionally different (2 structurally different start-up cases):
1. upon cold-start all PS's are at the same level (what is now assumed)
2. upon cold-start, a valid quorum of PS's are at the same level, but some of them PS's are behind with their log (think as if like ATS and all PSs would concurrently and non-deterministically be stopped at the same time)
  - in this case the ATS would instruct the left behind PS's to catch-up on a leader PS
  - this case is functionally different than 1. but can structurally be implemented the same as 1. (ie: 1. is a particular case of this)
3. hot-start of a PS into a running cluster

In light of this explanation, the `if (log_Gl.hdr.append_lsa == catchup_lsa)` condition above fits for situation 2) where the other up-to-date PSs that are equal to the leader PS are sent the catch-up command.